docker_guide.md

```diff
@@ -0,0 +1,97 @@
+# mozilla-mobile Docker Guide
+TaskCluster runs our automation tasks, such as compilation and unit tests, in a [Docker container](https://en.wikipedia.org/wiki/Docker_(software)). We maintain our own Docker images:
+- [`mozillamobile/focus-android`](https://hub.docker.com/r/mozillamobile/focus-android/)
+- [`mozillamobile/firefox-tv`](https://hub.docker.com/r/mozillamobile/firefox-tv/)
+
+ These images contain all the tools (e.g. Android SDK, Python runtime) needed to run our tasks. These images are built from a Dockerfile located in our repository (e.g. [focus-android Dockerfile](https://github.com/mozilla-mobile/focus-android/blob/master/tools/docker/Dockerfile)). Taskcluster downloads this image from [Docker Hub](https://hub.docker.com/) whenever it needs to run one of our tasks.
+
+If you want to build or use an image locally then you need to install the Community Edition of Docker on your machine:
+https://www.docker.com/community-edition
+
+## Updating the Docker image
+If our toolchain changes or if we require new tools for running tasks then it might be needed to update the image taskcluster uses. This sample will use the `mozillamobile/focus-android` docker image as an example:
+
+1. Install Docker Community Edition on your machine
+
+2. Modify [Dockerfile](https://github.com/mozilla-mobile/focus-android/blob/master/tools/docker/Dockerfile). See [Dockerfile reference](https://docs.docker.com/engine/reference/builder/) for a list of commands that can be used to build an image.
+
+3. Build the Dockerfile locally and tag it _mozillamobile/focus-android_:
+
+```bash
+docker build -t mozillamobile/focus-android .
+```
+
+4. Run the image locally and test that everything is working as you intended to:
+
+```bash
+docker run -it mozillamobile/focus-android /bin/bash
+```
+
+5. Commit Dockerfile and open a PR for it.
+
+6. Push the image to Docker Hub so that taskcluster can find it:
+
+```bash
+docker push mozillamobile/focus-android.
+```
```

Comment:
We should update this to mention versioning the image. That's what we do in android-components and it prevents overriding a working image with a broken one and subsequently breaking all automation tasks. Of course we do not need to do that now and could also just file an issue here..

In a nutshell:

```bash
# Tag the created image with a version number:
docker tag mozillamobile/focus-android:latest mozillamobile/focus-android:1.0

# Then push both tags
docker push mozillamobile/focus-android:latest
docker push mozillamobile/focus-android:1.0

# In the .taskcluster.yml make sure to specific the image with the version number. With that you can test a new image in a PR before using it for all other tasks.

Reply:
I filed #4.